src/header.tsx

```diff
@@ -155,66 +163,73 @@ export type HeaderMenuItemProps = JSX.IntrinsicElements["li"] & {
+    const config = useAppkitConfig();
+    const focusBgColor = useColorScheme().isHighContrast
+      ? config.colors.neutral90
+      : config.colors.neutral10;
+    const focusTextColor = useColorScheme().isHighContrast
+      ? config.colors.neutral05
+      : config.colors.neutral90;
 
-  const css = {
-    display: "flex",
-    gap: 16,
-    alignItems: "center",
-    minWidth: 160,
-    padding: 12,
-    textDecoration: "none",
-    color: config.colors.neutral90,
-    cursor: "pointer",
-    whiteSpace: "nowrap",
-    "& > svg": {
-      maxHeight: 23,
-      fontSize: 23,
-      width: 24,
-      strokeWidth: 2,
-      "& > path": { strokeWidth: "inherit" },
-    },
-    ":hover, :focus": {
-      backgroundColor: focusBgColor,
-      color: focusTextColor,
-    },
-    ...focusStyle(config, { inset: true }),
-  } as const;
+    const css = {
+      display: "flex",
+      gap: 16,
+      alignItems: "center",
+      minWidth: 160,
+      padding: 12,
+      textDecoration: "none",
+      color: config.colors.neutral90,
+      cursor: "pointer",
+      whiteSpace: "nowrap",
+      "& > svg": {
+        maxHeight: 23,
+        fontSize: 23,
+        width: 24,
+        strokeWidth: 2,
+        "& > path": { strokeWidth: "inherit" },
+      },
+      ":hover, :focus": {
+        backgroundColor: focusBgColor,
+        color: focusTextColor,
+      },
+      ...focusStyle(config, { inset: true }),
+    } as const;
 
-  const { className, ...outerProps } = rest;
-  const wrapperElem = wrapper ?? <></>;
-  return (
-    <li
-      role="menuitem"
-      {...outerProps}
-      css={{
-        ":not(:first-of-type)": {
-          borderTop: `1px solid ${config.colors.neutral30}`,
-        },
-        ...!wrapper && css,
-      }}
-      {...!wrapper && { className }}
-    >
-      {jsx(wrapperElem.type, {
-        key: wrapperElem.key,
-        ...wrapperElem.props,
-        children: <>
-          {icon ?? <svg />}
-          <div>{children}</div>
-        </>,
-        ...wrapper && { className },
-        ...wrapper && { css: [css, { borderRadius: "inherit" }] },
-      })}
-    </li>
-  );
-};
+    const { className, ...outerProps } = rest;
+    const wrapperElem = wrapper ?? <></>;
+    return (
+      <li
+        role="menuitem"
+        {...outerProps}
+        css={{
+          ":not(:first-of-type)": {
+            borderTop: `1px solid ${config.colors.neutral30}`,
+          },
+          ...!wrapper && css,
+        }}
+        {...!wrapper && {
+          className,
+          ref: r => typeof ref === "function" ? ref(r) : ref ? ref.current = r : {},
+        }}
+      >
+        {jsx(wrapperElem.type, {
+          key: wrapperElem.key,
+          ...wrapperElem.props,
+          children: <>
+            {icon ?? <svg />}
+            <div>{children}</div>
+          </>,
+          ...wrapper && {
+            className,
+            ref: (r: HTMLElement | null) => typeof ref === "function" ? ref(r) : ref ? ref.current = r : {},
```

Comment:
See above. Also: is the type annotation necessary here, given that it's not necessary above?

Reply:
Apparently it is 🤨. Without, it complains about implicitly having  an `any` type.

Reply:
Hu, well, then lets keep it :D